sympy/core/tests/test_numbers.py

```diff
@@ -1749,3 +1748,43 @@ def test_Integer_precision():
     assert Float('1.0', precision=Integer(15))._prec == 15
     assert type(Float('1.0', precision=Integer(15))._prec) == int
     assert sympify(srepr(Float('1.0', precision=15))) == Float('1.0', precision=15)
+
+def test_numpy():
+    from sympy.external import import_module
+    from sympy.utilities.pytest import skip
+    np = import_module('numpy')
+    
+    if not np:
+        skip('numpy not installed. Abort numpy tests.')
+
+    def equal(x, y):
+        return x == y and type(x) == type(y)
+    print(type(S(123)))
+    print(type(Float(np.int_(123))))
+    try:
+        assert equal(
+            Float(np.int_(12345678912345678)), S(12345678912345678.))
+        assert equal(
+            Float(np.intp(12345678912345678)), S(12345678912345678.))
+    except OverflowError:
+        # May fail on 32-bit systems: Python int too large to convert to C long
+        pass
+    assert equal(Float(np.intc(1234567891)), S(1234567891.))
+    assert equal(Float(np.int8(-123)), S(-123.))
+    assert equal(Float(np.int16(-12345)), S(-12345.))
+    assert equal(Float(np.int32(-1234567891)), S(-1234567891.))
+    assert equal(
+        Float(np.int64(-12345678912345678)), S(-12345678912345678.))
+    assert equal(Float(np.uint8(123)), S(123.))
+    assert equal(Float(np.uint16(12345)), S(12345.))
+    assert equal(Float(np.uint32(1234567891)), S(1234567891.))
+    assert equal(
+        Float(np.uint64(1234567891234567891)), S(1234567891234567891.))
+    assert equal(Float(np.float32(1.123456)), Float(1.123456, precision=24))
+    assert equal(Float(np.float64(1.1234567891234)),
+                 Float(1.1234567891234, precision=53))
+    assert equal(Float(np.longdouble(1.123456789)),
+                 Float(1.123456789, precision=80))
+    assert equal(Float(np.complex64(1 + 2j)), S(1.0 + 2.0 * I))
+    assert equal(Float(np.complex128(1 + 2j)), S(1.0 + 2.0 * I))
+    assert equal(Float(np.longcomplex(1 + 2j)), S(1.0 + 2.0 * I))
```

Comment:
These should test the precision. Also, I don't know if there are discrepancies in any of these types across platforms, but if there are, we should guard against them.

Reply:
Actually, it seems kind of odd that Float should accept a complex number. `Float(1j)` currently doesn't work. I think these should actually be raising an exception, at least until https://github.com/sympy/sympy/pull/12192 is finished. You can test that something raises an exception with `raises`.

Reply:
To test the precision we would need Float instead of S on the RHS but as @asmeurer noticed Float does not currently accept complex numbers. It does accept them through numpy-to-sympy converting function though - so maybe add a raise in Float there whenever np.complex gets passed as an argument?

Reply:
Added raise for np.complex64 and np.complex128.